tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+                finalBeamInfo = self.cco[beam]
+                if pupilOffset is None:
+                    desiredPupilPos = pupilPos
+                else:
+                    desiredPupilPos = np.add(pupilPos, pupilOffset)
+                if focalPlaneOffset is None:
+                    desiredFocalPlanePos = focalPlanePos
+                else:
+                    desiredFocalPlanePos = np.add(focalPlanePos, focalPlaneOffset)
+                self.assertPairsAlmostEqual(finalBeamInfo.pupilPos, desiredPupilPos, self.maxPupilPosErr)
+                self.assertPairsAlmostEqual(finalBeamInfo.focalPlanePos, desiredFocalPlanePos,
+                                            maxDiff=self.maxFocalPlanePosErr)
+
+    def testSampleBasics(self):
+        """Test basic elements of the sample coordinate converter
+        """
+        self.assertEqual(len(self.cco.cameraGeom), len(self.scc.detectorFracPosList))
+        self.assertEqual(list(self.cco.cameraGeom.getNameIter()),
+                         ["D" + str(i) for i in range(len(self.scc.detectorFracPosList))])
+        self.assertEqual(len(self.cco), 6)
+        self.assertEqual(tuple(self.cco.beamNames), ("0", "1", "2", "3", "4", "5"))
+        self.assertEqual(self.cco.maskInfo.numHoles, 6)
+        self.assertPairsAlmostEqual(self.cco.maskInfo.getHolePos("0"), (0, 0))
+        for beamInfo, beam in zip(self.cco, self.cco.beamNames):
+            self.assertEqual(beamInfo.name, beam)
+
+    def testSetPupilFieldAngleTrivial(self):
+        """Test setPupilFieldAngle for the trivial case of hole 0
+        aimed perpendicular to the center of the pupil
+        """
+        self.cco.setPupilFieldAngle(pupilPos=(0, 0))
+
+        # the telescope should be pointed at the center of the CBP and vice-versa
+        # NOTE: It would be nice to get better than the 0.0028" that I measure
+        self.assertSpherePointsAlmostEqual(self.cco.telAzAltInternal,
+                                           SpherePoint(Vector3d(*self.cco.config.cbpPosition)),
+                                           maxSep=0.01*arcseconds)
+        self.assertSpherePointsAlmostEqual(self.cco.cbpAzAltInternal,
+                                           SpherePoint(Vector3d(*(-self.cco.config.cbpPosition))),
+                                           maxSep=0.01*arcseconds)
+
+        self.assertAnglesAlmostEqual(self.cco.telRotInternal, 0*degrees)
+
+        # beam 0 should be pointed to the center of the pupil, normal to the pupil
+        # and land on the center of the focal plane and the center of detector D0
+        beamInfo0 = self.cco["0"]
+        self.assertEqual(beamInfo0.name, "0")
+        self.assertPairsAlmostEqual(beamInfo0.holePos, (0, 0))
+        self.assertFalse(beamInfo0.isOnPupil)  # blocked by the central obscuration
+        self.assertTrue(beamInfo0.isOnFocalPlane)
+        self.assertTrue(beamInfo0.isOnDetector)
+        self.assertPairsAlmostEqual(beamInfo0.focalPlanePos, (0, 0))
```

Comment:
Looks like I missed maxDiff in a lot of places, and had local values and a few numbers instead of the class attributes. I've fixed all of those that I found.